src/Compilers/CSharp/Portable/Parser/CharacterInfo.cs

```diff
@@ -196,9 +198,9 @@ public static bool IsValidIdentifier(string name)
         /// characters will have been dropped from the search string).
         /// See DevDiv #14432 for more.
         /// </summary>
-        internal static bool ContainsDroppedIdentifierCharacters(string name)
+        internal static bool ContainsDroppedIdentifierCharacters(string? name)
         {
-            if (string.IsNullOrEmpty(name))
+            if (RoslynString.IsNullOrEmpty(name))
```

Comment:
Is this code from before we moved to `netcoreapp3.1`? The use of `RoslynString` shouldn't be necessary in new annotations. If it is that would likely be a bug somewhere.

Reply:
Moving back to string.IsNullOrEmpty has broken the tests (`name might be null in 'name[0]'` below)

---
In reply to: [386037976](https://github.com/dotnet/roslyn/pull/42025#discussion_r386037976) [](ancestors = 386037976)

Reply:
@jaredpar This is strange. Maybe we can take a look at it tomorrow together?

Reply:
For the record: this file is being used in a project that targets .NET 2.0 framework, but where we enabled nullability. That is the reason for the diagnostic when just using `string.IsNullOrEmpty` (which lacks the flow analysis attribute on .NET 2.0, obviously).